### ISSUE TITLE
gb generate behaves oddly/errors in my project

### ISSUE THREAD

**Issue by PSG-Luna** [constabulary/gb#492]
When running gb generate, I get odd errors about two packages in the cmd/ subfolder, but the error paths refer to my actual GOPATH instead of the gb project dir.

Additionally, gb build doesn't ever rebuild those two packages unless I manually delete the bin/ folder to force a rebuild

Output of gb generate:

```
luna@dev:~/src/gb/delta$ env DEBUG=. gb generate
2015/12/15 11:23:31 cmd.go:88: project root "/home/luna/src/gb/delta"
2015/12/15 11:23:31 context.go:325: matchPackages: /home/luna/src/gb/delta/src ...
2015/12/15 11:23:31 main.go:139: args: [cmd/cdn-bucket cmd/model-migrate login model model/appengine model/cloud model/migration web/cdn web/gocode web/login web/main]
can't load package: package cmd/cdn-bucket: open /home/luna/src/golang-src/src/cmd/cdn-bucket: no such file or directory
can't load package: package cmd/model-migrate: open /home/luna/src/golang-src/src/cmd/model-migrate: no such file or directory
FATAL: command "generate" failed: exit status 1
2015/12/15 11:23:32 context.go:201: removing work directory: /tmp/gb043705555
```

**Comment by davecheney** [comment]
Is the source for the generate online ?

**Comment by davecheney** [comment]
> Is the source for the generate online ?

I'm an idiot, gb generate is a gb command. Show's how much I've used it.

**Comment by davecheney** [comment]
@PSG-Luna are there any symlinks involved here ?

**Comment by davecheney** [comment]
Oh, I think I see what is going on here. It's related to #416; our old friend `cmd/`

**Comment by davecheney** [comment]
I won't be able to fix this for the moment. The workaround is to _not_ place your source inside directory called `cmd` at the root of your project. ie

$PROJECT/src/cmd/foo <<< NOT OK
$PROJECT/src/github.com/psg-luna/project/cmd/foo << OK

Duplicate of #416

**Comment by davecheney** [comment]
I think I have a fix for this.

**Comment by davecheney** [comment]
Fixed in #511
